src/Hedgehog.Experimental/Gen.fs

```diff
@@ -382,154 +397,158 @@ module GenX =
-            autoInner<'a list> config recursionDepths
-            |> Gen.map Set.ofList
-            |> wrap}
-
-    | Shape.FSharpMap s ->
-        s.Accept {
-          new IFSharpMapVisitor<Gen<'a>> with
-          member __.Visit<'k, 'v when 'k : comparison> () =
-            autoInner<('k * 'v) list> config recursionDepths
-            |> Gen.map Map.ofList
-            |> wrap }
-
-    | Shape.Tuple (:? ShapeTuple<'a> as shape) ->
-        let eGens =
-          shape.Elements
-          |> Array.map mkRandomMember
-
-        gen {
-          let mutable target = shape.CreateUninitialized ()
-          for eg in eGens do
-            let! u = eg
-            target <- u target
-          return target
-        }
-
-    | Shape.FSharpRecord (:? ShapeFSharpRecord<'a> as shape) ->
-        let fieldGen =
-          shape.Fields
-          |> Array.map mkRandomMember
-
-        gen {
-          let mutable target = shape.CreateUninitialized ()
-          for eg in fieldGen do
-            let! u = eg
-            target <- u target
-          return target
-        }
-
-    | Shape.FSharpUnion (:? ShapeFSharpUnion<'a> as shape) ->
-        let caseFieldGen =
-          shape.UnionCases
-          |> Array.map (fun uc -> uc.Fields |> Array.map mkRandomMember)
-
-        gen {
-          let! tag = Gen.integral <| Range.constant 0 (caseFieldGen.Length - 1)
-          let mutable u = shape.UnionCases.[tag].CreateUninitialized ()
-          for f in caseFieldGen.[tag] do
-            let! uf = f
-            u <- uf u
-          return u
-        }
-
-    | Shape.Enum _ ->
-        let values = Enum.GetValues(typeof<'a>)
-        gen {
-          let! index = Gen.integral <| Range.constant 0 (values.Length - 1)
-          return values.GetValue index |> unbox
-        }
-
-    | Shape.CliMutable (:? ShapeCliMutable<'a> as shape) ->
-        let propGen = shape.Properties |> Array.map mkRandomMember
-        gen {
-          let mutable target = shape.CreateUninitialized ()
-          for ep in propGen do
-            let! up = ep
-            target <- up target
-          return target
-        }
-
-    | Shape.Poco (:? ShapePoco<'a> as shape) ->
-        let bestCtor =
-          shape.Constructors
-          |> Seq.filter (fun c -> c.IsPublic)
-          |> Seq.sortBy (fun c -> c.Arity)
-          |> Seq.tryHead
-
-        match bestCtor with
-        | None -> failwithf "Class %O lacking an appropriate ctor" typeof<'a>
-        | Some ctor ->
-          ctor.Accept {
-          new IConstructorVisitor<'a, Gen<'a>> with
-            member __.Visit<'CtorParams> (ctor : ShapeConstructor<'a, 'CtorParams>) =
-              let paramGen = autoInner<'CtorParams> config recursionDepths
-              gen { 
-                let! args = paramGen
-                return ctor.Invoke args
+    match config.Overrides.Wrapped.TryFind typeof<'a>.FullName with
+    | Some gen -> gen |> Gen.map unbox<'a> |> wrap
```

Comment:
What is `wrap` doing there?  If `'b` is `'a`, is `wrap` just the identity function?  The code still compiles and passes all tests if I remove it.  I am guessing that it is a copy-paste from the other cases below.

![2020-12-13_20-46-20_423](https://user-images.githubusercontent.com/34664007/102034745-6be3ca00-3d84-11eb-9a25-bf73560a57eb.png)

Reply:
IIRC it's from before I took over the auto-gen code (which still fundamentally works the same was as back then). At the time there was a lot of stuff there I didn't understand, so I just left it. I've never understood `wrap`, but never tried removing it.

Reply:
I'm confused by what you mean by

> If `'b` is `'a`

Are they the same?

Also, I tried removing `wrap`, but it won't compile.

Reply:
The tooltip says `'b is 'a`, so my "if `'b` is `'a`" was meant to be rhetorical.  Sorry for being unclear.

I will try removing `unwrap` again later.

Reply:
Continuing this discussion in PR #46